fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -425,6 +425,10 @@ public FHIRPersistenceJDBCImpl(Properties configProps, IConnectionProvider cp) t
                         return resultBuilder.success(false).build();
                     }
                 }
+                
+                if (searchContext.getPageSize() == 0) {
+                    searchContext.setSummaryParameter(SummaryValueSet.COUNT);
+                }
```

Comment:
This will add _summary=count to the self(search) link of the result.

Reply:
I think it's okay as it is implicit behavior per the spec but I'll defer to @lmsurpre for his thoughts.

Reply:
I also think it's OK if this is aligned with the spec.

Reply:
I changed it to check pageSize before executing the resource query instead of setting the summary parameter.

Reply:
FWIW, I have no problem with this setting _summary=count in the self link.  I'm fine with whatever got merged.